src/connectivity/mod.rs

```diff
@@ -323,23 +323,39 @@ pub fn is_weakly_connected(graph: &digraph::PyDiGraph) -> PyResult<bool> {
 ///     for all edges.
 /// :param float default_weight: If ``weight_fn`` is not used this can be
 ///     optionally used to specify a default weight to use for all edges.
+/// :param float null_value: An optional float that will treated as a null
+///     value. If any element in the input matrix is this value it will be
+///     treated as not an edge. By default this is ``0.0``.
 ///
 ///  :return: The adjacency matrix for the input dag as a numpy array
 ///  :rtype: numpy.ndarray
-#[pyfunction(default_weight = "1.0")]
-#[pyo3(text_signature = "(graph, /, weight_fn=None, default_weight=1.0)")]
+#[pyfunction(default_weight = "1.0", null_value = "0.0")]
+#[pyo3(
+    text_signature = "(graph, /, weight_fn=None, default_weight=1.0, null_value=0.0)"
+)]
 pub fn digraph_adjacency_matrix(
     py: Python,
     graph: &digraph::PyDiGraph,
     weight_fn: Option<PyObject>,
     default_weight: f64,
+    null_value: f64,
 ) -> PyResult<PyObject> {
     let n = graph.node_count();
-    let mut matrix = Array2::<f64>::zeros((n, n));
+    let mut matrix = if null_value != 0.0 {
+        Array2::<f64>::from_elem((n, n), null_value)
+    } else {
+        Array2::<f64>::zeros((n, n))
+    };
```

Comment:
Is `zeros` much more efficient? Would this work?
```suggestion
    let mut matrix = Array2::<f64>::from_elem((n, n), null_value);
```

Reply:
I don't know if it's more efficient but I think it should work. I'll go with it. I doubt initializing is more expensive than allocating an nxn array anyway